From 8e7c5fa8593ba75a409d45da1d8a442e0592e74e Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 22 Oct 2024 18:09:34 +0200
Subject: [PATCH 1464/1695] plat-stm32mp1: stm32mp1_stpmic1: remove use of
 shared_resource driver

Remove use of shared_resource driver from platform stm32mp1 driver
since secure/non-secure assignation of resources and validation
of the platform consistency is now managed using the firewall
framework. This change is a step in the removal of the now deprecated
shared_resources driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a33937dbeef6ac0b5841540750baf29adb5a116b plat-stm32mp1: stm32mp1_stpmic1: remove use of shared_resource driver]
Change-Id: I75f0dad0cbf7c1c106b583df55f9b28d715c655c
Link: https://github.com/OP-TEE/optee_os/commit/dd6b0423e5907bb86f66d8fc1b1dd3621fffe7a5
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/413523
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../plat-stm32mp1/drivers/stm32mp1_stpmic1.c  | 27 -------------------
 1 file changed, 27 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c
index b3e4d9594..75f9f3c10 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c
@@ -761,28 +761,6 @@ void stm32mp_put_pmic(void)
 		stm32_i2c_suspend(i2c_handle);
 }
 
-static void register_non_secure_pmic(void)
-{
-	/* Allow this function to be called when STPMIC1 not used */
-	if (!i2c_handle->base.pa)
-		return;
-
-	stm32mp_register_non_secure_pinctrl(i2c_handle->pinctrl);
-	if (i2c_handle->pinctrl_sleep)
-		stm32mp_register_non_secure_pinctrl(i2c_handle->pinctrl_sleep);
-
-	stm32mp_register_non_secure_periph_iomem(i2c_handle->base.pa);
-}
-
-static void register_secure_pmic(void)
-{
-	stm32mp_register_secure_pinctrl(i2c_handle->pinctrl);
-	if (i2c_handle->pinctrl_sleep)
-		stm32mp_register_secure_pinctrl(i2c_handle->pinctrl_sleep);
-
-	stm32mp_register_secure_periph_iomem(i2c_handle->base.pa);
-}
-
 static void init_pmic_secure_state(void)
 {
 	if (i2c_handle->i2c_secure)
@@ -809,11 +787,6 @@ static TEE_Result initialize_pmic(const void *fdt, int pmic_node)
 	DMSG("PMIC version = 0x%02lx", pmic_version);
 	stm32mp_put_pmic();
 
-	if (pmic_is_secure())
-		register_secure_pmic();
-	else
-		register_non_secure_pmic();
-
 	parse_regulator_fdt_nodes(fdt, pmic_node);
 
 	return TEE_SUCCESS;
-- 
2.47.2

